### DISCUSSION TITLE
show only my own suppliers

### DISCUSSION THREAD

**Discussion by xu4wang** [directus/directus#13344] [Archive]
I have three models:
- suppliers
- buyers
- contracts

One supplier can have multiple buyers and one buyer can buy from multiple supplier.  One contract in contracts has one supplier and one buyer.

How can I set up the access control rule to make the buyer can only see the suppliers related to him/her?

The logic is that for each contract in contracts, if the buyer is the current user, show the related supplier in the suppliers collection.
I am not sure is it possible to achieve by Directus APP configuration?

Thanks,Austin

**Comment by rijkvanzanten** [comment]
> How can I set up the access control rule to make the buyer can only see the suppliers related to him/her?


Just to clarify, do you mean "buyer can only see `suppliers` where the `buyer` has a `contract` with a `supplier`" ?

**Reply by xu4wang** [reply]
Hi @rijkvanzanten Yes. Exactly.  

I meant  for all the suppliers, a buyer can only see a subset of suppliers where the buyer has a contract with.

**Reply by rijkvanzanten** [reply]
Assuming `suppliers` also has a m2m field to `contracts`, and I'm assuming that each `buyer` has a Directus User associated through a `user` m2o. The other field names are just guesses I have based on the limited schema info supplied. You should be able to configure permissions for `suppliers` to something like

```json
{
  "contracts": {
    "buyer_id": {
      "user_id": {
        "_eq": "$CURRENT_USER"
      }
    }
  }
}
```

**Reply by david-zacharias** [reply]
We have a similar setup with `harbours` and `audits` connected as M2M via `harbour_audits`. `audits` have a field for selecting a system user. Goal is to show a logged on user only harbours where an audit exists where he is assigned to.

We use this rule in the related user group on the `harbours` collection:

```json
"audits": {
  "audit_id": {
    "auditor": {
      "id": {
        "_eq": "$CURRENT_USER"
      }
    }
  }
}
```

In directus 9.10.0 this worked as expected and described in @rijkvanzanten previous for the `contracts` example.

However, with upgrade to directus 9.11.0 the rule did not restrict the results to harbours a user is related to via an audit anymore. Instead all harbours of the `harbours`-collection are shown in the UI as long as at least one audit exists which the current user is assigned to. The actual relation between the harbour and the audit is no longer respected.

I have not had the time to dive into the changes made with 9.11.0 to think of what may changed this behaviour (unintentionally).

**Reply by azrikahar** [reply]
@david-zacharias can you try upgrading to 9.11.1 to see whether the issue still persist? The rule not behaving as intended may be related to #13469 (happens on 9.11.0, but was fixed on 9.11.1)

**Reply by david-zacharias** [reply]
The system I have the above mentioned configuration on, runs with a google cloud database which can not be used with directus 9.1.11 because auf [this issue](https://github.com/directus/directus/issues/13535) 😬 But I am happy to check with the next release and report back, if this is okay for you

**Reply by azrikahar** [reply]
> The system I have the above mentioned configuration on, runs with a google cloud database which can not be used with directus 9.1.11 because auf [this issue](https://github.com/directus/directus/issues/13535) 😬 But I am happy to check with the next release and report back, if this is okay for you

No worries! Hope it'll be resolved by then 👍

**Reply by david-zacharias** [reply]
Just tested with 9.12.0 and everything works as expected ❤️
